src/idl_gen_general.cpp

```diff
@@ -731,6 +731,7 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
        it != struct_def.fields.vec.end();
        ++it) {
     auto &field = **it;
+    assert(field.value.type.base_type != BASE_TYPE_ARRAY);
```

Comment:
asserts just produce a crash for the user. if people start making schemas with arrays in them, then unsupported languages need to get a proper error message somehow.